src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -1214,6 +1215,32 @@ private fun checkDuplicates(
     message.addToHolder(holder)
 }
 
+private fun checkConstGenerics(holder: RsAnnotationHolder, constParameter: RsConstParameter) {
+    val availability = CONST_GENERICS.availability(constParameter)
+    if (availability == AVAILABLE) return
+
+    val version = constParameter.cargoProject?.rustcInfo?.version
+    val feature = when (constParameter.typeReference?.type) {
+        is TyInteger, is TyBool, is TyChar -> {
+            val current = version?.semver
+            val since = MIN_CONST_GENERICS.since
+            if (current != null && since != null && current >= since) MIN_CONST_GENERICS else CONST_GENERICS
```

Comment:
```suggestion
            if (current != null && since != null && current.isGreaterOrEqualThan(since.major, since.minor, since.patch)) MIN_CONST_GENERICS else CONST_GENERICS
```

Also, the same fix we need to add into `org.rust.lang.core.CompilerFeature#availability` to support case when you use `1.51.0-nightly` and feature requires at least `1.51.0`. In this case, we shouldn't show any error annotation

Reply:
If you know better fix here, you are welcome